src/legacy/server/config/get_unused_config_keys.test.ts

```diff
@@ -0,0 +1,274 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { LegacyPluginSpec } from '../../../core/server/legacy/plugins/find_legacy_plugin_specs';
+// @ts-ignore
+import { transformDeprecations } from './transform_deprecations';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { LegacyConfig } from '../../../core/server/legacy/config';
+import { getUnusedConfigKeys } from './get_unused_config_keys';
+
+jest.mock('./transform_deprecations', () => ({
+  transformDeprecations: jest.fn().mockImplementation(s => s),
+}));
+
+describe('getUnusedConfigKeys', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    transformDeprecations.mockImplementation((s: any) => s);
+  });
+
+  const getConfig = (values: Record<string, any> = {}): LegacyConfig =>
+    ({
+      get: () => values as any,
+    } as LegacyConfig);
+
+  describe('not using core or plugin specs', () => {
+    it('should return an empty list for empty parameters', async () => {
+      expect(await getUnusedConfigKeys([], [], [], {}, getConfig())).toEqual([]);
+    });
+
+    it('returns empty list when config and settings have the same properties', async () => {
+      expect(
+        await getUnusedConfigKeys(
+          [],
```

Comment:
I agree, this will improve readability. Will do. `settings` and `config` params should also probably be renamed. Took me some time to understand what was the config definition and what was the actual input parameters